index.bs

```diff
@@ -5483,6 +5500,377 @@ At this time, one [=credential property=] is defined: the [=resident key credent
 :: None.
 
 
+# User Agent Automation # {#sctn-automation}
+
+For the purposes of user agent automation and [=web application=] testing, this document defines a number of [[WebDriver]] [=extension commands=].
+
+## <dfn>Virtual Authenticators</dfn> ## {#sctn-automation-virtual-authenticators}
+
+These WebDriver [=extension commands=] create and interact with [=Virtual Authenticators=]: software implementations of the
+<a>Authenticator Model</a>. <a>Virtual Authenticators</a> are stored in a <dfn>Virtual Authenticator Database</dfn>.
+Each stored [=virtual authenticator=] has the following properties:
+
+: <dfn>authenticatorId</dfn>
+:: An non-null string made using up to 48 characters from the `unreserved` production defined in Appendix A of [[RFC3986]]
+    that uniquely identifies the [=Virtual Authenticator=].
+: |protocol|
+:: The protocol the [=Virtual Authenticator=] speaks: either `ctap2` or `ctap1/u2f` [[FIDO-CTAP]].
+: |transport|
+:: The {{AuthenticatorTransport}} simulated. If the |transport| is set to {{AuthenticatorTransport/internal}}, the
+    authenticator simulates [=platform attachment=]. Otherwise, it simulates [=cross-platform attachment=].
+: |hasResidentKey|
+:: If set to [TRUE] the authenticator will support [=resident credentials=].
+: |hasUserVerification|
+:: If set to [TRUE], the authenticator supports [=user verification=].
+: |isUserPresent|
+:: Determines the result of a [=test of user presence=] performed on the [=Virtual Authenticator=]. If set to [TRUE],
+    a [=test of user presence=] will always succeed. If set to [FALSE], it will fail.
```

Comment:
I assume this also defines the outcome of any other kind of request for user consent? Should that then be stated explicitly?

Reply:
Well, hm, a subtle p[oint there -- it seems from the "Flags" table a the very end of section [5.2. authenticatorGetAssertion (0x02)](https://drafts.fidoalliance.org/fido-2/stable-links-to-latest/fido-client-to-authenticator-protocol.html#authenticatorGetAssertion) that it is possible for `uv` to be returned true and `up` false, and the definition of [User Consent](https://www.w3.org/TR/webauthn-2/#user-consent) punts to  [authorization gesture](https://www.w3.org/TR/webauthn-2/#authorization-gesture) which say it could be one or the other.

Reply:
Hm, good point. Maybe leaving it explicitly coupled to just `up` is a better way to go after all?